### PR TITLE
[sidelining] remove `check_only_add_effects`

### LINKED ISSUE #980: clean up check_only_add_effects
we don't actually need it anymore

### PR DESCRIPTION
**Pull request by NishanthJKumar**
Removes unnecessary + confusing checks that were previously being done in the `_find_best_matching_pnad_and_sub` function. These were mostly vestigial from when we used to be doing specialization. As a result, this PR also removes certain unit tests that were only testing this vestigial functionality.
Closes #980

### CODE REVIEW

**Review by tomsilver — APPROVED** [review]
thx!

**Review by ronuchit — CHANGES_REQUESTED** [review]

File: src/nsrt_learning/strips_learning/base_strips_learner.py [hunk 1]
```diff
@@ -264,30 +263,28 @@ def _find_best_matching_pnad_and_sub(
...
-                                                      segment.init_atoms)
                     if not next_atoms.issubset(segment.final_atoms):
                         continue
                     # If the segment has a non-None necessary_add_effects,
                     # and the ground operator's add effects don't fit this,
                     # skip.
-                    if segment.necessary_add_effects is not None and \
-                       not segment.necessary_add_effects.issubset(
-                           ground_op.add_effects):
+                    if not check_only_preconditions and \
```
[the remark this answers is not in the document]
Comment (NishanthJKumar) [reply src/nsrt_learning/strips_learning/base_strips_learner.py:276]: good point!

File: src/nsrt_learning/strips_learning/base_strips_learner.py [hunk 2]
```diff
@@ -264,30 +264,27 @@ def _find_best_matching_pnad_and_sub(
...
-                                                      segment.init_atoms)
                     if not next_atoms.issubset(segment.final_atoms):
                         continue
                     # If the segment has a non-None necessary_add_effects,
                     # and the ground operator's add effects don't fit this,
                     # skip.
                     if segment.necessary_add_effects is not None and \
-                       not segment.necessary_add_effects.issubset(
-                           ground_op.add_effects):
+                        not segment.necessary_add_effects.issubset(
```
Comment (ronuchit) [inline src/nsrt_learning/strips_learning/base_strips_learner.py:278]: revert this line and the next line?

File: src/nsrt_learning/strips_learning/base_strips_learner.py [hunk 3]
```diff
@@ -264,30 +264,27 @@ def _find_best_matching_pnad_and_sub(
...
+                            ground_op.add_effects):
                         continue
+                else:
+                    # If check_only_preconditions is True, we must be
+                    # calling this from spawning during backchaining
+                    # with a most-general PNAD that has no add effects
+                    # and all other atoms sidelined, and thus this
```
Comment (ronuchit) [inline src/nsrt_learning/strips_learning/base_strips_learner.py:285]: atoms -> predicates

**Review by ronuchit — APPROVED** [review]
lgtm, minor finishing touches!

**Pull request merged**